Objective-C/CBLDocument.mm

```diff
@@ -127,6 +128,16 @@ - (instancetype) initWithDatabase: (CBLDatabase*)database
     return self;
 }
 
+- (instancetype) initWithCollection: (CBLCollection*)collection
+                         documentID: (NSString*)documentID
+                     includeDeleted: (BOOL)includeDeleted
+                              error: (NSError**)outError {
+    return [self initWithDatabase: collection.db
```

Comment:
The collection need to be retained in the document and the database used in the CBLDocument will need to get from the collection. This means that any constructors that has database as a parameter should be changed to constructor.

Reply:
> should be changed to constructor.

should be changed to collection.